docs/source/contributing/development.md

```diff
@@ -0,0 +1,450 @@
+
+- Follow the Pythonic `snake_case` naming convention when you define methods.
+- Ensure each such public method is
+  [type annotated](https://docs.python.org/3/library/typing.html) and is well
+  documented with a docstring following the
+  [standard](https://github.com/robocorp/rpaframework/issues/357#issue-1100365552)
+  format:
+  - Summary describing what the keyword does.
+  - Parameters description.
+  - Python and Robot Framework usage examples.
+- Ensure each method is low-coupled to the others and contain high-cohesive code
+  blocks.
+- While not necessary, you can make these stand out by using the `@keyword` decorator (
+  imported with `from robot.api.deco import keyword`), place where you can configure
+  a custom name for the expected keyword name (if the default is not satisfactory) and
+  additional settings like `tags` and `types`.
+- Hiding keywords logging due to security/privacy reasons can be achieved easily with
+  the `RPA.Robocorp.utils.protect_keywords` utility.
+
+### Library structure
+
+Usually all these methods go into this single main library class, but sometimes you get
+to write a lot of them and is not desirable to manage thousands of lines of code in a
+single module. So you'd normally split the initial module into multiple such modules
+which treat different topics and groups closely related methods into individual keyword
+classes. Then you'd take all these "library parts" and unite them together into the
+main library class.
+
+Some good examples with slightly different approaches:
+- Composition (has-a): [RPA.DocumentAI](../../../packages/main/src/RPA/DocumentAI/DocumentAI.py)
+- Inheritance (is-a): [RPA.Desktop](../../../packages/main/src/RPA/Desktop/__init__.py)
+
+So based on your needs, feel free to choose the simplest and most Pythonic one.
+
+#### Separate package (if needed)
+
+Some libraries are too havy on dependencies and the number of keywords to be
+implemented directly into [main](../../../packages/main), therefore we create a new
+package in this "monolith" repo, and we place them inside by following the same
+structure. Afterwards this becomes self-contained and gets released as a separate
+package into PyPI.
+
+Based on the frequency of usage (and slightly approachable  dependency train), such
+package might get automatically included in the _main_ one  by pinning it into the
+[pyproject.toml](../../../packages/main/pyproject.toml) file. If not, then for sure
+this has to be pinned in the _meta_'s [pyproject.toml](../../../pyproject.toml) in
+order to get documentation built and test how all our packages will go altogether.
+
+### Required changes
+
+Is not easy to know by heart all the changes that need to be done when improving or
+adding a new library, so here's a list with all the areas of interest:
+- The library module/package itself discussed above.
+- Updates into _pyproject.toml_:
+  - Its newly required dependencies should be added with `poetry add <package>` command.
```

Comment:
`-d` is being deprecated (or is gone now) in our version of Poetry. They use groups now, so the command would be `poetry add <package> -G dev`. I also suggest showing how to pin a version so perhaps the full example should be `poetry add <package>[@^<version>] [-G dev]`?